Lib/asyncio/windows_events.py

```diff
@@ -308,6 +308,22 @@ def __init__(self, proactor=None):
             proactor = IocpProactor()
         super().__init__(proactor)
 
+    def run_forever(self):
+        try:
+            # if _self_reading_future is cancelled -
+            # this might indicate that event loop was interrupted before
+            # and self-reading routine is not hooked up now and needs
+            # to be restarted
+            if (self._self_reading_future is not None and
+                self._self_reading_future.cancelled()):
```

Comment:
Please replace `.cancelled()` with `.done()`.
Interruption can occur when the `recv()` called `fut.set_result()` but before `loop._loop_self_reading()` callback executed on the next loop iteration.